ngraph/core/reference/include/ngraph/runtime/reference/non_max_suppression.hpp

```diff
@@ -0,0 +1,52 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <functional>
+#include <map>
+#include "ngraph/coordinate_transform.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/shape_util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            void non_max_suppression(const float* boxes_data,
```

Comment:
Does input data always have float?

Reply:
Before calling this function from `op::v5::NonMaxSuppression::evaluate`, there is a preprocessing in lines 1069-1087. After above mentioned preprocessing, `boxes_data` are floats.